src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java

```diff
@@ -1281,38 +1281,72 @@ public Group createGroup(JSONObject groupJson)
+            channel = null;
 
-        final JSONObject guildObject = object.getJSONObject("guild");
+            final String groupName = channelObject.optString("name");
+            final long groupId = channelObject.getLong("id");
+            final String groupIconId = channelObject.optString("icon", null);
 
-        final String guildIconId = guildObject.optString("icon", null);
-        final long guildId = guildObject.getLong("id");
-        final String guildName = guildObject.getString("name");
-        final String guildSplashId = guildObject.optString("splash", null);
-        final VerificationLevel guildVerificationLevel = VerificationLevel.fromKey(Helpers.optInt(guildObject, "verification_level", -1));
-        final int presenceCount = Helpers.optInt(object, "approximate_presence_count", -1);
-        final int memberCount = Helpers.optInt(object, "approximate_member_count", -1);
-        final Set<String> guildFeatures;
+            final JSONArray usernameArray = channelObject.optJSONArray("recipients");
+            final List<String> usernames;
+            if (usernameArray == null)
+                usernames = null;
+            else
+                usernames = Collections.unmodifiableList(StreamSupport.stream(usernameArray.spliterator(), false).map(String::valueOf).collect(Collectors.toList()));
 
-        if (guildObject.isNull("features"))
+            group = new InviteImpl.GroupImpl(groupIconId, groupName, groupId, usernames);
+        }
+        else if (channelType == ChannelType.TEXT || channelType == ChannelType.VOICE)
         {
-            guildFeatures = Collections.emptySet();
+            type = Invite.Type.GUILD;
+
+            final JSONObject guildObject = object.getJSONObject("guild");
+
+            final String guildIconId = guildObject.optString("icon", null);
+            final long guildId = guildObject.getLong("id");
+            final String guildName = guildObject.getString("name");
+            final String guildSplashId = guildObject.optString("splash", null);
+            final VerificationLevel guildVerificationLevel = VerificationLevel.fromKey(Helpers.optInt(guildObject, "verification_level", -1));
+            final int presenceCount = Helpers.optInt(object, "approximate_presence_count", -1);
+            final int memberCount = Helpers.optInt(object, "approximate_member_count", -1);
+
+            final Set<String> guildFeatures;
+            if (guildObject.isNull("features"))
+                guildFeatures = Collections.emptySet();
+            else
+                guildFeatures = Collections.unmodifiableSet(StreamSupport.stream(guildObject.getJSONArray("features").spliterator(), false).map(String::valueOf).collect(Collectors.toSet()));
+
+            guild = new InviteImpl.GuildImpl(guildId, guildIconId, guildName, guildSplashId, guildVerificationLevel, presenceCount, memberCount, guildFeatures);
+
+            final String channelName = channelObject.getString("name");
```

Comment:
Why are you using `optString` for the name when you handle the name for group and `getString` here?

Reply:
I just left in there what we used previously. It makes sense though because groups don't always have a name, channels in a guild however are required to have one.